styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java

```diff
@@ -37,7 +38,7 @@
 
   R triggerExecution(@Getter WorkflowInstance workflowInstance, Trigger trigger);
   R info(@Getter WorkflowInstance workflowInstance, Message message);
-  R dequeue(@Getter WorkflowInstance workflowInstance);
+  R dequeue(@Getter WorkflowInstance workflowInstance, Set<String> resourceIds);
```

Comment:
This was `@Nullable` to handle backwards compatibility consistently, like in the `submitted` event below, which has `@Nullable String executionId`.

Reply:
I think it is usually a good practice using empty collection instead of null to indicate no value. But this case is different.

Reply:
Yeah I'd like to be able to use empty collection (no `null`s and no `Optionals` too); it would be a great move to make but let's make it consistently, for all events that currently have backwards compatibility concerns and resolve them with @Nullable fields for distinguishing "no field even existed" from "no value".

Reply:
Apologies for going back and forth; actually I agree with making it non-null already (equating "no resources were used" with "we don't know what resources were used") because it's ok that the scheduler behaves the same, for all purposes AFAICT...